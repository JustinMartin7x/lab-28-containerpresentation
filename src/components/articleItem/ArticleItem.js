import React from 'react';
import PropTypes from 'prop-types';

export default function ArticleItem({ article }) {
  return (
    <div>
      <h1>{article.title}</h1>
      <p>{article.author}</p>
      <p>{article.content}</p>
    </div>
  );
}

ArticleItem.propTypes = {
  article: PropTypes.object,
};
