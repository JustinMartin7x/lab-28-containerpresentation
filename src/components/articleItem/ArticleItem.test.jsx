import React from 'react';
import { render, cleanup } from '@testing-library/react';
import ArticleItem from '../articleItem/ArticleItem';



describe('Article item component', () => {
  afterEach(() => cleanup());
  it ('renders an article', () => {
    const { asFragment } = render(<ArticleItem
      article = {{
        title: 'hello', 
        author: 'some author',
        description: 'something else described..'
      }}
    />);

    expect(asFragment()).toMatchSnapshot();
  });
});

