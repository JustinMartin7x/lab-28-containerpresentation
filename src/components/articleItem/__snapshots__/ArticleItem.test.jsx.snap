// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article item component renders an article 1`] = `
<DocumentFragment>
  <div>
    <h1>
      hello
    </h1>
    <p>
      some author
    </p>
    <p />
  </div>
</DocumentFragment>
`;
